modal/client.py

```diff
@@ -262,8 +250,8 @@ async def verify(cls, server_url: str, credentials: tuple[str, str]) -> None:
         """mdmd:hidden
         Check whether can the client can connect to this server with these credentials; raise if not.
         """
-        async with cls(server_url, api_pb2.CLIENT_TYPE_CLIENT, credentials):
-            pass  # Will call ClientHello RPC and possibly raise AuthError or ConnectionError
+        async with cls(server_url, api_pb2.CLIENT_TYPE_CLIENT, credentials) as client:
+            client.hello()  # Will call ClientHello RPC and possibly raise AuthError or ConnectionError
```

Comment:
I guess we could also have just gotten rid of the `hello` method and moved the logic in here if it's the only place we still use it.

Reply:
I was thinking about it and I think it's actually useful to keep the `hello` method indefinitely as a simple ping method – it's useful for tests and verifying tokens. I might revert some of the recent changes and add back logic for the warnings etc, just in a way where it's not a part of client construction.